docs/dev/checks/python/check_api.md

```diff
@@ -1,73 +1,61 @@
-# API
+# Anatomy of a Check
 
-
-## AgentCheck parent class
-
-Every check is a subclass of `AgentCheck` and must implement the `check` function:
+An Agent Check is a Python class that inherits from `AgentCheck` and implements
+the `check` method:
 
 ```python
 from checks import AgentCheck
 
 class MyCheck(AgentCheck):
     def check(self, instance):
         # Do the check. Collect metrics, emit events, submit service checks,
-        [..]
+        # ...
 ```
 
-The Agent instantiates a new `MyCheck` for each instance it finds in
-`mycheck.yaml`, passing that instance's configuration options to the `check`
-function via the `instance` parameter. The `check` method will be called on each
-[collector][collector] run to collect metrics.
+The Agent creates an object of type `MyCheck` for each element contained in the
+`instances` sequence within the corresponding config file:
 
-The check inherits some useful attributes from the `AgentCheck` parent class:
+```
+instances:
+  - host: localhost
+    port: 6379
 
-- `self.name`: the name of the check
-- `self.init_config`: the `init_config` section from the check configuration.
-- `self.log`: a [logger](https://docs.python.org/2/library/logging.html).
+  - host: example.com
+    port: 6379
+```
 
-You can use these anywhere in the check function.
+Any mapping contained in `instances` is passed to the `check` method through the
+named parameter `instance`. The `check` method will be invoked at every run of the
+[collector][collector].
 
-If a check cannot run because of improper configuration, programming error or
-because it could not collect any metrics, it should raise a meaningful
-exception. This exception will be logged, as well as being shown in the Agent info
-command for easy debugging.
+The `AgentCheck` base class provides some useful attributes and methods:
 
-Do not override the `run` and `get_warnings` methods from `AgentCheck` unless you
-know exactly what you're doing.
+- `self.name`: the name of the check
+- `self.init_config`: the `init_config` section from the check configuration.
+- `self.log`: a [logger](https://docs.python.org/2/library/logging.html).
+- `get_instance_proxy()`: a function returning a dictionary containing informations on the Proxy being used
 
-## Overriding the \_\_init\_\_ method
+**Warning**: when loading a Python check, the Agent will ignore any class that
+has been derived - **you should never derive from an existing Check**.
```

Comment:
This seems a little misleading, because the Agent doesn't strictly ignored derived classes, it ignores intermediate base classes... could we change this to reflect that?